packages/web3torrent/src/components/torrent-info/wires-list/WiresList.tsx

```diff
@@ -0,0 +1,56 @@
+import prettier from 'prettier-bytes';
+import React from 'react';
+import {Torrent} from '../../../types';
+import './WiresList.scss';
+import {ChannelState} from '../../../clients/payment-channel-client';
+import {PaidStreamingWire} from '../../../library/types';
+
+export type UploadInfoProps = {
+  torrent: Torrent;
+  channelCache: Record<string, ChannelState>;
+  channelIds: string[];
+  peerType: 'seeder' | 'leecher';
+};
+
+const WiresList: React.FC<UploadInfoProps> = ({torrent, channelCache, channelIds, peerType}) => {
+  function wireToTableRow(wire: PaidStreamingWire) {
+    const peer = wire.paidStreamingExtension.peerAccount;
+    let channelId: string;
+    if (peerType === 'seeder') {
+      channelId = wire.paidStreamingExtension.pseChannelId;
+    } else {
+      channelId = wire.paidStreamingExtension.peerChannelId;
+    }
+    return (
+      channelIds.includes(channelId) && (
+        <tr className="peerInfo" key={peer}>
+          <td>
+            <button>Close</button>
+          </td>
+          <td className="channel-id">{channelId}</td>
+          <td className="peer-id">{peer}</td>
+          <td className="downloaded">
+            {prettier(wire.uploaded)}
+            &nbsp;
+            {peerType == 'seeder' ? `up` : `down`}
+          </td>
+          {peerType == 'seeder' ? (
+            <td className="earned">{Number(channelCache[channelId].beneficiaryBalance)} wei</td>
+          ) : (
+            <td className="paid">-{Number(channelCache[channelId].beneficiaryBalance)} wei</td>
+          )}
+        </tr>
+      )
+    );
+  }
+
+  return (
+    <section className="wires-list">
+      <table className="wires-list-table">
+        <tbody>{Object.values(torrent.wires).map(wireToTableRow)}</tbody>
+      </table>
```

Comment:
I've just realised an issue with this approach. When we finish downloading, it seems as if the wire is thrown away: that means it will disappear from the UI. The channel still exists, of course, and we want to be able to click "Close" so we can extract our money. 

A solution would be to have a "channels first" table, with the "downloaded/uploaded bytes" field grabbed with a reverse lookup through the existing wires. Then, if the wire is destroyed we will only lose this field (and we could then worry about saving the information somewhere so we never lose it).

Reply:
I like the `channels first` approach. Otherwise we have to build another UI for users to close channels and extract money.